nucliadb/src/nucliadb/ingest/service/writer.py

```diff
@@ -121,6 +121,7 @@ async def NewKnowledgeBox(  # type: ignore
                 description=request.config.description,
                 semantic_model=semantic_model,
                 release_channel=release_channel,
+                external_index_provider=request.external_index_provider,
```

Comment:
As it is right now, we'll have the default with type `UNSET` but a default pinecone config in the oneof field, as backend is not populating anything

Reply:
this is always the case with protobuf. the concept of optional fields isn't really supported, they will always default to the default value of the primitive fields.

This is why having a type can help and make the code much cleaner:
- first check the type, then access the corresponding config.